python/social_bot/envs/gazebo_base.py

```diff
@@ -133,6 +133,48 @@ def render(self, mode='human'):
     def close(self):
         super().close()
         gazebo.close_without_model_base_fini()
+    
+    def insert_model(self, model, name=None, pose="0 0 0 0 0 0"):
+        """
+        Insert a model with into a specific position of the world
+        Args:
+            model (string): the name of the model in the model database
+            name (string): the name of the model in the world
+            pose (string): the pose of the model, format is "x y z roll pitch yaw"
+        """
+        if name == None:
+            name = model
+        model_sdf = """
+        <?xml version='1.0'?>
+        <sdf version ='1.4'>
+        <model name=""" + name + """>
+            <include>
+                <uri>model://""" + model + """</uri>
+            </include>
+            <pose frame=''>""" + pose + """</pose>
+        </model>
+        </sdf>
+        """
+        self._world.insertModelFromSdfString(model_sdf)
+        # Sleep for a while waiting for gzserver to finish the inserting
+        # operation. Or it may not be successfully inserted and report error.
+        time.sleep(0.2)
```

Comment:
Not directly related to this change: If goal is not reached in max_steps (100 or 200 in some tasks), then we'll have to wait 0.2 second?  Would that be too much overhead?  Would it be possible to step first, and catch the exception if it occurs?  Also, have anyone tested with fewer than 20 steps to save some time?

Reply:
It's fine. Only during initializing will there be inserting operation. Reset
 does not need that.
I have spend a lot time on the insertion problem, increase the sleeping and stepping little by little... this is safe for most case.

Reply:
Sounds good, thanks for explaining!